tabbycat/standings/metrics.py

```diff
@@ -107,10 +107,7 @@ def get_annotated_queryset(self, queryset, round=None):
     def annotate_with_queryset(self, queryset, standings):
         """Annotates items with the given QuerySet."""
         for item in queryset:
-            metric = getattr(item, self.key)
-            if metric is None:
-                metric = 0
-            standings.add_metric(item, self.key, metric)
```

Comment:
Just a heads-up, I think it might have been written like this because `item[self.key]` sometimes exists and is `None`. For example, if the field in question is a `SUM` aggregation but no records match the relevant filter, then I think the SQL field might be populated as `NULL` (as opposed to 0, the sum of one or more zeroes). This can also happen in things like variance, when there's one record (because to compute variance meaningfully you need at least two). PostgreSQL or other behaviour might've changed since then, but this should be checked and noted to be sure. (Also, this really should have been commented with the reason at the time—I think that was me, sorry!)

I tried it quickly, to reproduce just mark four teams as unavailable before generating the round:
![image](https://user-images.githubusercontent.com/1725499/116770845-5271b080-a9fb-11eb-801e-f0188659bb65.png)

I think this used to crash the standings generator because Python can't compare `NoneType` to `int`. But that problem presumably disappeared because you moved ranking annotations to SQL, right?

Still wanted to give a heads-up about how it shows though—this might mean there's an inconsistency between "team exists but has no results" ("None") and "can't find the metric" (0).

Reply:
So for SQL-based rankings (with all the ranking precedence in a single query), nulls are handled (see 7a1b68c94f229998e45e2c91acf65c9163a5eb53) and sorting is included, so Python's sorting is not called on "`None`-able" fields. In the general case where Python's sorting is required, I added a default of 0 in the metric getter in order to handle `None`s. I think it would be more valuable to show that such info is missing than show 0.

Good point for the inconsistency. Changed to give `None`.